kura/org.eclipse.kura.core.comm/src/main/java/org/eclipse/kura/core/comm/CommConnectionImpl.java

```diff
@@ -287,16 +289,16 @@ private synchronized ByteBuffer getResponse(int timeout, int demark) throws IOEx
         return buffer.limit() > 0 ? buffer : null;
     }
 
-    private String getBytesAsString(byte[] bytes) {
+    private static String getBytesAsString(byte[] bytes) {
         if (bytes == null) {
             return null;
-        } else {
-            StringBuffer sb = new StringBuffer();
-            for (byte b : bytes) {
-                sb.append("0x").append(Integer.toHexString(b)).append(" ");
-            }
+        }
 
-            return sb.toString();
+        StringBuffer sb = new StringBuffer(bytes.length * 3);
```

Comment:
Out of curiosity, is there any specific reason of preferring StringBuffer to StringBuilder?